### DISCUSSION TITLE
Upgraded to 2.6.2 but some lookup on log QSO not working

### DISCUSSION THREAD

**Discussion by sipsmi** [magicbug/Cloudlog#2917] [General]
Yes I have cleared the cache.

Normally when logging a QSO the system shows previous QSOs with that user and looks up name in QRZ.  This works for some calls I have worked with but since 2.6.2 I have issues logging/lookups for many callsigns.   The error log seems to suggest an SQL error (see below).  Any pointers to what may be going on here?

This seems to be mainly an issue with callsigns I have not worked before.  If I try logging a call I have previously worked all the fields fill in and the QRZ/HAMQTH icons appear next to the call.

Cheers

Si G0FOZ

```
ERROR - 2024-01-10 13:42:27 --> Query error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1 - Invalid query: select name, cqz, lat, 'long' from dxcc_entities where adif = 
ERROR - 2024-01-10 13:42:27 --> Severity: error --> Exception: Call to a member function result() on bool /home/si/public_html/log/application/models/Logbook_model.php 4196
 
```

**Comment by magicbug** [comment]
Do you by chance know what MySQL version your using?

**Reply by sipsmi** [reply]
mysql  Ver 8.0.35-0ubuntu0.22.04.1 for Linux on x86_64 ((Ubuntu))

**Reply by sipsmi** [reply]
for some reason the $dxcc parameter passed to "function get_entity($dxcc)" in Logbook_mode.php is null for these call instances

**Reply by magicbug** [reply]
I don't think it will fix it but can you go to Admin -> Update Country Files, once that completes can you try log a QSO with me "2M0SQL" and let me know if same error happens.

**Reply by sipsmi** [reply]
First thing I tried when I saw it was something to do with dxcc lookup :-(
Same error with your callsign :-(

**Comment by sipsmi** [comment]
Just to keep working I added a line -          if ( ! isset($dxcc) || empty($dxcc)) { return '' ; } - so the code does not fail and this allows to log calls I have not seen before - will dig deeper when I get a minute.

  ```
  public function get_entity($dxcc){
            # temp kludge to work around issue of null dxcc for unworked callsign on log QSO
            #
           if ( ! isset($dxcc) || empty($dxcc)) { return '' ; }
            # end kludge
      $sql = "select name, cqz, lat, 'long' from dxcc_entities where adif = " . $dxcc;
      $query = $this->db->query($sql);

      if ($query->result() > 0){
          $row = $query->row_array();
          return $row;
      }
      return '';
    }

```

**Comment by R1BLH** [comment]
@magicbug 
> once that completes can you try log a QSO with me "2M0SQL" and let me know if same error happens.

If Cloudlog is configured to search hamqth.com, everything works fine -- information and photos are displayed.
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/ab13c86a-c578-47f7-aeb2-aecc69ae6485)

If Cloudlog is configured to search qrz.com for information, nothing is displayed.
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/6ecc95ef-a6d1-4e41-8d1a-1f9b85fbaa5e)

If you enter a non-existent callsign, it will say that it was not found.
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/52463e66-afab-4a4e-9740-aba1ba04d9db)

**Reply by R1BLH** [reply]
@magicbug 
When Cloudlog is configured to use qrz.com for automatic lookup, the mysqli error is logged when filling in the new QSO form fields:

`ERROR - 2024-03-04 21:58:22 --> Severity: error --> Exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1 /var/www/r1blh.cats-home.net/log/system/database/drivers/mysqli/mysqli_driver.php 301
`

Server Software | Apache/2.4.41 (Ubuntu)
PHP Version | 8.1.27
MySQL Version | 10.3.39-MariaDB-0ubuntu0.20.04.2

**Reply by R1BLH** [reply]
The same situation is on the Cloudlog v.2.6.6. Upgrading the MariaDB server to the latest version (11.3.2-MariaDB-1:11.3.2+maria~ubu2004) did not give any result. PHP version is 8.1.27 (the same situation is with the PHP v.7.4.33).

![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/9689d8e4-a0f1-45ec-bb97-3a6919ee1c63)

**Reply by sipsmi** [reply]
My hack earlier in this thread stops the problem from occurring ( only on lookups for calls I have not worked before ).

The file is:  /application/models/Logbook_model.php

```
public function get_entity($dxcc){
          # temp kludge to work around issue of null dxcc for unworked callsign on log QSO
          #
         if ( ! isset($dxcc) || empty($dxcc)) { return '' ; }
          # end kludge
```

**Reply by R1BLH** [reply]
This does not work in my case, neither on php 7.4, nor on php 8.1.

**Reply by substa** [reply]
Same problem here, fix by @sipsmi doesn't work for me.

Server Software	Apache/2.4.57 (Debian)
PHP Version	8.2.12
MySQL Version	8.3.0

**Comment by peacekiller** [comment]
I ran into the same issue after I updated cloudlofg from 2.5.? to 2.6.14. So I tried updating the country informations but this hasn't solved the problem. Also clearing the browser cache has no effect. I changed the lookup callbook from 'qrz' to 'hamqth' an the issue disappeared for know. I looking forward this issue will be solved soon. thank you for your work.

Maybe some additional helpful informations:
```
ERROR - 2024-06-03 18:24:24 --> Severity: error --> Exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1 /opt/cloudlog/system/database/drivers/mysqli/mysqli_driver.php 303

ERROR - 2024-06-03 18:26:07 --> SQL failed: select name, cqz, lat, 'long' from dxcc_entities where adif =  error: mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1 in /opt/cloudlog/system/database/drivers/mysqli/mysqli_driver.php:301
Stack trace:
#0 /opt/cloudlog/system/database/drivers/mysqli/mysqli_driver.php(301): mysqli->query()
#1 /opt/cloudlog/system/database/DB_driver.php(730): CI_DB_mysqli_driver->_execute()
#2 /opt/cloudlog/system/database/DB_driver.php(620): CI_DB_driver->simple_query()
#3 /opt/cloudlog/application/models/Logbook_model.php(4337): CI_DB_driver->query()
#4 /opt/cloudlog/application/controllers/Logbook.php(840): Logbook_model->get_entity()
#5 /opt/cloudlog/application/controllers/Logbook.php(142): Logbook->partial()
#6 /opt/cloudlog/system/core/CodeIgniter.php(485): Logbook->json()
#7 /opt/cloudlog/index.php(316): require_once('...')
#8 {main}

```

Cloudlog Version 2.6.14
PHP Version	8.1.2-1ubuntu2.17
MySQL Version	5.5.5-10.6.16-MariaDB-0ubuntu0.22.04.1
